sklearn/neighbors/_base.py

```diff
@@ -351,31 +354,33 @@ def _check_algorithm_metric(self):
         if self.algorithm not in ["auto", "brute", "kd_tree", "ball_tree"]:
             raise ValueError("unrecognized algorithm: '%s'" % self.algorithm)
 
+        self._metric = self.metric
```

Comment:
I could be missing something. Why do we define `_metric` here?

Reply:
We define it here, because we do not want to modify `self.metric` here, and we also do not want to expose `self.metric_`.

Reply:
I am wrong: it was needed but is not needed anymore.

afdaaa1f5b7f4cbc6c568ae64ad08b1413c0d1ef removes it and its commit message explains why.